src/Workspaces/CSharp/Portable/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -69,6 +69,7 @@ public override AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken p
                             !currentToken.IsParenInArgumentList() &&
                             !currentToken.IsDotInMemberAccess() &&
                             !currentToken.IsCloseParenInStatement() &&
+                            !currentToken.IsCloseParenInTuple() &&
```

Comment:
i'm a little as to what's going on here.  looking at this code, it says: if teh previous token is `}`, but it's not hte `}` of an expression , and the current token is not the `)` of a tuple expression....  then do this thing.

This implies that previously we had a case where we had a `}` as the previous token and the current token was the `)` of a tuple expression...

But:
1. i don't see how that could happen.
2. i don't see how the test reflects that case.

Can you clarify?

Reply:
n/m.  i see.  i think the problem is this .    whne you have `a => { }`, then that `}` *should* be considered the "close brace of an expression" right?  So the right fix IMO would be to fix IsCloseBraceOfExpression right?

Reply:
Good point, I guess so. Looks like that that _could_ simplify some more logic here. I'm going to look into that, now. Thnx!

Reply:
absolutely!  also, if you need to chat more, you can come to gitter or discord.  lots of help available there!